modules/tensor_mechanics/src/actions/DomainIntegralAction.C

```diff
@@ -65,6 +65,18 @@ DomainIntegralAction::validParams()
       "displacements",
       "The displacements appropriate for the simulation geometry and coordinate system");
   params.addParam<VariableName>("temperature", "", "The temperature");
+  params.addParam<VariableName>(
+      "youngs_modulus_derivative",
+      "",
+      "A grading parameter for cracks in functinonally graded materials (optional). Must be "
+      "provided as an auxiliary "
+      "variable that captures an exponential function that transitions the modulus of elasticity "
+      "from the value of a material to another one.");
+  params.addParam<VariableName>(
+      "space_dependent_youngs_modulus",
```

Comment:
I have a couple of questions about how the user should define these. I'll come discuss.